### PR TITLE
don't sign X.509 certs

### PR DESCRIPTION
**Pull request by lijunwangs**
#### Problem

This is to resurrect PR from https://github.com/solana-labs/solana/pull/34202 to address merge conflicts and some comments.

#### Summary of Changes
This get rid of 3rd party components rcgen in the path of private key access to make the code more secure.

Fixes #

### CODE REVIEW

**Comment by lijunwangs** [comment]
> Can you push your commits directly to #34202 so we have all the context and discussions in one place? These are some guides on how to do that: https://tighten.com/insights/adding-commits-to-a-pull-request/ https://stackoverflow.com/questions/22237609/

Unfortunately, I do not have permission to push to the firedancer repo. So the technique would not work.

The real change is c1bf6c0e5d1cb56e63bba38a862cb6d4178cc751 -- rest is just address merge conflicts.


+    // See https://www.itu.int/rec/T-REC-X.509-201910-I/en for detailed definitions.
 
     let mut cert_der = Vec::<u8>::with_capacity(0xf4);
     //    Certificate SEQUENCE (3 elem)
@@ -53,13 +54,13 @@ pub fn new_dummy_x509_certificate(keypair: &Keypair) -> (rustls::Certificate, ru
     //            algorithm OBJECT IDENTIFIER 1.3.101.112 curveEd25519 (EdDSA 25519 signature algorithm)
     //          subjectPublicKey BIT STRING (256 bit)
     cert_der.extend_from_slice(&[
-        0x30, 0x81, 0xf1, 0x30, 0x81, 0xa4, 0xa0, 0x03, 0x02, 0x01, 0x02, 0x02, 0x08, 0x01, 0x01,
-        0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x30, 0x11,
-        0x31, 0x0f, 0x30, 0x0d, 0x06, 0x03, 0x55, 0x04, 0x03, 0x0c, 0x06, 0x53, 0x6f, 0x6c, 0x61,
-        0x6e, 0x61, 0x30, 0x20, 0x17, 0x0d, 0x37, 0x30, 0x30, 0x31, 0x30, 0x31, 0x30, 0x30, 0x30,
-        0x30, 0x30, 0x30, 0x5a, 0x18, 0x0f, 0x34, 0x30, 0x39, 0x36, 0x30, 0x31, 0x30, 0x31, 0x30,
-        0x30, 0x30, 0x30, 0x30, 0x30, 0x5a, 0x30, 0x00, 0x30, 0x2a, 0x30, 0x05, 0x06, 0x03, 0x2b,
-        0x65, 0x70, 0x03, 0x21, 0x00,
+        0x30, 0x81, 0xf6, 0x30, 0x81, 0xa9, 0xa0, 0x03, 0x02, 0x01, 0x02, 0x02, 0x08, 0x01, 0x01,
+        0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x30, 0x16,
+        0x31, 0x14, 0x30, 0x12, 0x06, 0x03, 0x55, 0x04, 0x03, 0x0c, 0x0b, 0x53, 0x6f, 0x6c, 0x61,
+        0x6e, 0x61, 0x20, 0x6e, 0x6f, 0x64, 0x65, 0x30, 0x20, 0x17, 0x0d, 0x37, 0x30, 0x30, 0x31,
+        0x30, 0x31, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x5a, 0x18, 0x0f, 0x34, 0x30, 0x39, 0x36,
+        0x30, 0x31, 0x30, 0x31, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x5a, 0x30, 0x00, 0x30, 0x2a,
+        0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x03, 0x21, 0x00,

Which is just restore the issuer common name back to "Solana node" from "Solana"

File: client/src/connection_cache.rs [hunk 1]
```diff
@@ -253,7 +251,7 @@ mod tests {
     #[test]
     fn test_connection_with_specified_client_endpoint() {
         // Start a response receiver:
-        let (response_recv_socket, response_recv_exit, keypair2, response_recv_ip) = server_args();
+        let (response_recv_socket, response_recv_exit, keypair2, _) = server_args();
```
Comment (behzadnouri) [inline client/src/connection_cache.rs:254]: If it is not used anywhere, should `server_args()` be updated not to return `response_recv_ip`?

**Review by behzadnouri — APPROVED** [review]
lgtm,
Can you also plz comment on https://github.com/solana-labs/solana/pull/34202 pointing to this one for reference and close that as well?

File: client/src/connection_cache.rs (hunk 1 above)
Comment (lijunwangs) [reply client/src/connection_cache.rs]: Fixed. Thanks!

**Review by behzadnouri — APPROVED** [review]
lgtm

**Comment by lijunwangs** [comment]
Just for record keeping, the CERT looks like the following:

openssl x509 -in dummy.cer -inform DER -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 72340172838076673 (0x101010101010101)
        Signature Algorithm: ED25519
        Issuer: CN = Solana node
        Validity
            Not Before: Jan  1 00:00:00 1970 GMT
            Not After : Jan  1 00:00:00 4096 GMT
        Subject: 
        Subject Public Key Info:
            Public Key Algorithm: ED25519
                ED25519 Public-Key:
                pub:
                    06:ab:db:0e:56:94:64:04:cc:ae:ab:56:87:55:da:
                    32:46:23:16:40:e7:ff:21:66:b8:23:94:7c:65:c1:
                    2f:1f
        X509v3 extensions:
            X509v3 Subject Alternative Name: critical
                DNS:localhost
            X509v3 Basic Constraints: critical
                CA:FALSE
    Signature Algorithm: ED25519
         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff

**Pull request merged**